teos/src/main.rs

```diff
@@ -64,11 +63,17 @@ fn create_new_tower_keypair(db: &DBM) -> (SecretKey, PublicKey) {
     db.store_tower_key(&sk).unwrap();
     (sk, pk)
 }
-
 #[tokio::main]
 async fn main() {
     let opt = Opt::from_args();
     let path = config::data_dir_absolute_path(opt.data_dir.clone());
+    let conf_file_path = path.join("teos.toml");
+
+    // Clone the conf_file_path variable before passing it to the from_file function
+    let cloned_conf_file_path = conf_file_path.clone();
```

Comment:
Cloning this is not necessary. Instead, you can make `config::from_file` receive `&Pathbuf` instead of `Pathbuf`

Reply:
> Cloning this is not necessary. Instead, you can make `config::from_file` receive `&Pathbuf` instead of `Pathbuf`

I did this `// Log config file path and any non-default options
    log::info!("Using configuration file: {:?}", &conf_file_path);` but it's showing error **borrow of moved value and does not implement copy trait**. That's the reason I cloned to reference it after the first one.

Reply:
> Cloning this is not necessary. Instead, you can make `config::from_file` receive `&Pathbuf` instead of `Pathbuf`

I tried with &Pathbuf but it says `expected struct Pathbuf, found &Pathbuf`

Reply:
> Cloning this is not necessary. Instead, you can make `config::from_file` receive `&Pathbuf` instead of `Pathbuf`

I did reference it but shows error

`let mut conf = config::from_file::<Config>(&conf_file_path);
    let is_default = conf.is_default();
    conf.patch_with_options(opt);
    conf.verify().unwrap_or_else(|e| {
        eprintln!("{e}");
        std::process::exit(1);
    });

    // log default data dir
    log::info!("Using default data directory: {:?}", &path);

    // Log config file path and any non-default options
    log::info!("Using configuration file: {:?}", &conf_file_path);`

Reply:
> Cloning this is not necessary. Instead, you can make `config::from_file` receive `&Pathbuf` instead of `Pathbuf`

If I log the config file path before config::from_file, it is not showing error. So, I did this first `log::info!("Using configuration file: {:?}", conf_file_path);` then `let mut conf = config::from_file::<Config>(conf_file_path);`. It is working now.